edx_sga/static/js/src/edx_sga.js

```diff
@@ -16,6 +16,7 @@ function StaffGradedAssignmentXBlock(runtime, element) {
         var removeGradeUrl = runtime.handlerUrl(element, 'remove_grade');
         var template = _.template($(element).find("#sga-tmpl").text());
         var gradingTemplate;
+        var studentAssignments;
```

Comment:
Pretty much yes. Before this change, each interaction with the page (meaning each click) resulted in a ajax call to the backend to re-fetch the entire set of results.
I'm not sure this is the best way to solve it, but it's a (quick?) fix for the performance issue here.